etc/docs/Design.md

```diff
@@ -0,0 +1,100 @@
+# Things to cover
+
+## API
+
+### High level Library API
+
+The library API should provide the following methods to the clients:
+
+* Initialize an executor instance with the following settings
+  * Limits for resources
+    
+
+* Execute a command with the supplied arguments, that returns a pointer to a `Command` object that can be later used to manipulate the underlying process
+
+### GRPC API definition
+
+The GRPC API defines the following methods:
+
+* ExecuteCommand: Starts a command with limited resources in its own namespace with the supplied arguments
+* StopExecution : Stop a certain execution 
```

Comment:
How will you stop a job?

Reply:
I plan to send a SIGTERM then wait a few seconds and send a SIGKILL